pkg/controllers/management/k3sbasedupgrade/k3sbased_upgrade_handler.go

```diff
@@ -157,7 +157,8 @@ func (h *handler) deployK3sBasedUpgradeController(cluster *mgmtv3.Cluster) error
 	case cluster.Status.Driver == mgmtv3.ClusterDriverRke2:
 		appName = Rke2AppName
 	}
-	app, err := appLister.Get(systemProjectName, appName)
+	systemProjectBackingNamespace := systemProject.Status.BackingNamespace
```

Comment:
Just to mention that in the case where the value is empty, `appLister.Get` will fail to find the app and return errors, which is acceptable.

Reply:
@JonCrowther , is call the function `systemProject.GetProjectBackingNamespace()` a better option here?